bindings/pyroot_experimental/PyROOT/src/RVecPyz.cxx

```diff
@@ -0,0 +1,123 @@
+   if (!pyinterface) {
+      PyErr_SetString(PyExc_RuntimeError, "Object not convertible: __array_interface__ does not exist.");
+      return NULL;
+   }
+   if (!PyDict_Check(pyinterface)) {
+      PyErr_SetString(PyExc_RuntimeError, "Object not convertible: __array_interface__ is not a dictionary.");
+      return NULL;
+   }
+
+   // Get the data-pointer
+   auto pydata = PyDict_GetItemString(pyinterface, "data");
+   if (!pydata) {
+      PyErr_SetString(PyExc_RuntimeError, "Object not convertible: __array_interface__['data'] does not exist.");
+      return NULL;
+   }
+   long data = PyLong_AsLong(PyTuple_GetItem(pydata, 0));
+
+   // Get the size of the contiguous memory
+   auto pyshape = PyDict_GetItemString(pyinterface, "shape");
+   if (!pyshape) {
+      PyErr_SetString(PyExc_RuntimeError, "Object not convertible: __array_interface__['shape'] does not exist.");
+      return NULL;
+   }
+   long size = 0;
+   for (unsigned int i = 0; i < PyTuple_Size(pyshape); i++) {
+      if (size == 0) size = 1;
+      size *= PyLong_AsLong(PyTuple_GetItem(pyshape, i));
+   }
+
+   // Get the typestring
+   auto pytypestr = PyDict_GetItemString(pyinterface, "typestr");
+   if (!pytypestr) {
+      PyErr_SetString(PyExc_RuntimeError, "Object not convertible: __array_interface__['typestr'] does not exist.");
+      return NULL;
+   }
+   std::string typestr = CPyCppyy_PyUnicode_AsString(pytypestr);
+   const auto length = typestr.length();
+   if(length != 3) {
+      PyErr_SetString(PyExc_RuntimeError,
+              ("Object not convertible: __array_interface__['typestr'] returned '" + typestr + "' with invalid length unequal 3.").c_str());
+      return NULL;
+   }
+
+   // Verify correct endianess
+   const auto endianess = typestr.substr(1, 2);
+#ifdef R__BYTESWAP
+   const auto byteswap = "<";
+#else
+   const auto byteswap = ">";
+#endif
+   if (!endianess.compare(byteswap)) {
+      PyErr_SetString(PyExc_RuntimeError, "Object not convertible: Endianess of __array_interface__['typestr'] does not match endianess of ROOT.");
+      return NULL;
+   }
+
+   const auto dtype = typestr.substr(1, length);
+   std::string cppdtype;
+   if (dtype == "i4") {
+      cppdtype = "int";
+   } else if (dtype == "u4") {
+      cppdtype = "unsigned int";
+   } else if (dtype == "i8") {
+      cppdtype = "long";
+   } else if (dtype == "u8") {
+      cppdtype = "unsigned long";
+   } else if (dtype == "f4") {
+      cppdtype = "float";
+   } else if (dtype == "f8") {
+      cppdtype = "double";
+   } else {
+      PyErr_SetString(PyExc_RuntimeError, ("Object not convertible: Python object has unknown data-type '" + dtype + "'.").c_str());
+      return NULL;
+   }
+
+   // Construct an RVec of the correct data-type
+   const std::string klassname = "ROOT::VecOps::RVec<" + cppdtype + ">";
+   auto address = (void*) gInterpreter->Calc("new " + klassname + "(reinterpret_cast<" + cppdtype + "*>(" + data + ")," + size + ")");
```

Comment:
Ok this took a while. We need to create the object with a `new` so that the interpreter does not clean it up at the end and produces a double delete since the GC already cleaned it up.

Reply:
So Python will own the object and the corresponding Python proxy will take care of cleaning it?

Reply:
Correct!